### PR TITLE
feat: expose integer Tutor version parts to templates

### PR DESCRIPTION
**Pull request by kdmccormick**
@michaelwheeler

### CODE REVIEW

**Comment by michaelwheeler** [comment]
For some context, this is related to a question that I brought up during the Tutor Users' Group about writing plugins that support multiple Tutor versions. We're currently doing something along the lines of...

```
{% if TUTOR_VERSION.startswith("14.") %}
# Do something for Tutor v14 only...
{% else %}
# Do something for Tutor versions after 14...
{% endif %}
```

But having major/minor/patch versions available to the context as integers would allow more precision and flexibility.

**Comment by regisb** [comment]
@michaelwheeler I'd like to understand better your use case. Have you considered having multiple branches for your plugin, one for every major release?

**Comment by michaelwheeler** [comment]
> Have you considered having multiple branches for your plugin, one for every major release? 

In our team, managing multiple branches for our plugins felt like more overhead than adding an occasional conditional. We also think it will slightly simplify the process of upgrading to a newer named release.

**Comment by arbrandes** [comment]
I think this would useful to have as an option for plugin authors, particularly for cases where the plugins don't change that much between releases.

**Comment by regisb** [comment]
Right. I'm still not sure if it makes sense to add these settings to Tutor core. You do know that you could define these variables in your own plugin by adding a callback to the `ENV_TEMPLATE_VARIABLES` filter, right?

**Comment by michaelwheeler** [comment]
> You do know that you could define these variables in your own plugin by adding a callback to the `ENV_TEMPLATE_VARIABLES` filter, right?

I didn't know that!

It still feels to me like Tutor should provide the variable, since the version is a property of Tutor itself, but if that's undesirable for other reasons then maybe `ENV_TEMPLATE_VARIABLES` can get me where I need to go.

File: tutor/env.py [hunk 1]
```diff
@@ -64,6 +64,9 @@ def _prepare_environment() -> None:
             ("HOST_USER_ID", utils.get_user_id()),
             ("TUTOR_APP", __app__.replace("-", "_")),
             ("TUTOR_VERSION", __version__),
+            ("TUTOR_VERSION_MAJOR", int(__version__.split('.')[0])),
+            ("TUTOR_VERSION_MINOR", int(__version__.split('.')[1]))
+            ("TUTOR_VERSION_PATCH", int(__version__.split('.')[2]))
```
Comment (kdmccormick) [inline tutor/env.py:69]:
```suggestion
            ("TUTOR_VERSION_MINOR", int(__version__.split('.')[1])),
            ("TUTOR_VERSION_PATCH", int(__version__.split('.')[2])),
```

**Comment by kdmccormick** [comment]
Sorry for the delay @michaelwheeler  I think this is ready to go. Can you review it and try it out with one of your plugins?

**Comment by michaelwheeler** [comment]
> I think this is ready to go. Can you review it and try it out with one of your plugins?

Yep, thanks! I'll try to take a look before the end of next week.

**Comment by michaelwheeler** [comment]
Hey @kdmccormick, I tested this out and things work as expected from my perspective. Thanks!

**Comment by kdmccormick** [comment]
@michaelwheeler Would you have any objection to me removing `TUTOR_VERSION_PATCH` from this PR? You'd still have `TUTOR_VERSION_MAJOR` and `TUTOR_VERSION_MINOR`, but the patch version would remain unavailable to plugin authors without manually parsing `TUTOR_VERSION`.

My thinking is that it would be nice, as maintainers, to have some flexibility with the patch version. We don't currently use alpha/beta suffixes or release candidates (`19.2.3a`, `19.2.rc1`, etc.) but I can't say for sure that we never will, so I'd rather not encourage plugin authors to rely on it always being an integer.

**Pull request merged**